src/docs/src/best-practices/jsdevel.rst

```diff
@@ -46,3 +46,254 @@ tips and tricks that will ease the difficulty.
+    }
+
+5. Object key order.
+
+   Object key order may change between versions, so any views which rely on
+   that order may emit different results depending on the engine version:
+
+.. code-block:: bash
+
+   % js
+   js> r={}; ["Xyz", "abc", 1].forEach(function(v) {r[v]=v;}); Object.keys(r)
+   ["Xyz", "abc", "1"]
+
+   % js91
+   js> r={}; ["Xyz", "abc", 1].forEach(function(v) {r[v]=v;}); Object.keys(r)
+   ["1", "Xyz", "abc"]
+
+Using QuickJS
+=============
+
+The QuickJS-based JavaScript engine is available as of CouchDB version 3.4. It
+has to be explicitly enabled via ``[couchdb] js_engine = quickjs`` and
+restarting the service.
+
+Generally, QuickJS engine is a bit faster, consumes less memory, and provides
+slightly better isolation between contexts by re-creating the whole javascript
+engine runtime on every ``reset`` command.
+
+To try building invidual views using QuickJS, even when the default engine is
+SpiderMonkey, can use the ``"javascript_quickjs"`` as the view language,
+instead of ``"javascript"``. Just that view will be rebuilt using the QuickJS
+engine. However, when switching back to ``"javascript"`` the view will have to
+be re-built again.
+
+QuickJS vs Spidermonkey incompatibilities
+============================================
+
+The QuickJS engine is quite compatible with Spidermonkey version 91. The same
+incompatibilities between 1.8.5 and 91 are also present between 1.8.5 and
+QuickJS. So, when switching from 1.8.5 to QuickJS see the ``Spidermonkey version
+compatibility`` section above.
+
+These are a few incompatibilties between Spidermonkey 91 and QuickJS engine:
+
+1. ``RegExp.$1``, ..., ``RegExp.$9``
+
+This is a deprecated JavaScript feature that's not available in QuickJS.
+https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/n
+
+2. ``Date.toString()`` doesn't include the timezone name, just the offset.
+
+.. code-block:: bash
+
+   % qjs > (new Date()).toString();
+   "Thu Sep 05 2024 17:03:23 GMT-0400"
+
+   % js91
+   js>  (new Date()).toString();
+   "Thu Sep 05 2024 17:04:03 GMT-0400 (EDT)"
+
+Scanning for QuickJS incompatibilities
+======================================
+
+CouchDB version 3.4 and higher include a background scanner which can be used
+traverse all the databases and design documents and run them agaiinst
+Spidermonkey and the QuickJS engine and report any discrepancies in the logs.
+That could be a useful run before deciding to switch to QuickJS as the default
+JavaScript engine.
+
+The scanner can be enabled with:
+
+.. code-block:: ini
+
+   [couch_scanner_plugins]
+   couch_quickjs_scanner_plugin = true
+
+And configured to run at a predetermined time or on a periodic schedule. For
+instance:
+
+.. code-block:: ini
+
+   [couch_quickjs_scanner_plugin]
+   after = 2024-09-05T18:10:00
+   repeat = 1_day
+
+It won't start until after the specified time and then it will run about once
```

Comment:
Shall we use contractions?  That is, "won't" should be spelled as "will not", I think.

Reply:
I don't mind, pun intended ;-), using contractions, but I'll fix it if we think it looks better in the docs without them.